soc/arm/st_stm32/stm32h7/Kconfig.series

```diff
@@ -20,3 +20,7 @@ config SOC_SERIES_STM32H7X
 config STM32H7_DUAL_CORE
 	bool "Dual Core"
 	depends on SOC_SERIES_STM32H7X
+
+config STM32H7_BOOT_M4_AT_INIT
+	bool "Boot M4 core during M7 init independent of option byte BCM4."
+	default y
```

Comment:
Ok, I see what is happening here.
Please use `soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series` then.
I'll take care same is done for `STM32H7_DUAL_CORE`

Reply:
@erwango This was quite successful. Do you want to move `STM32H7_DUAL_CORE` in a separate merge request or here? If it is just about the move I could also write a commit for that on this branch.

Reply:
>  @erwango This was quite successful. Do you want to move STM32H7_DUAL_CORE in a separate merge request or here? If it is just about the move I could also write a commit for that on this branch.

Would be great. Let use another PR, so this does get into the way of current PR

Reply:
Looking to compliance check, it seems this is not yet the good solution.
Can you have a try in soc/arm/st_stm32/stm32h7/Kconfig.soc ?

Reply:
Yes, this would probably work since it was the initial attempt which actually worked (and passed the CI I think). I force-pushed such a revision. But after reading the "Boards, SoCs, etc." paragraph from the [Board Porting Guide](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html) I understand why you initially voted to move this from `Kconfig.soc` to `Kconfig.series` since this is for sure not an option affecting a SOC but instead the whole STM32H7 family.

But placing it in `Kconfig.series` does not work because it is interpreted like a SOC selection option if I remember correctly.